packages/blade/src/utils/isPartialMatchObjectKeys/isPartialMatchObjectKeys.ts

```diff
@@ -0,0 +1,76 @@
+/* eslint-disable no-console */
+import isObject from 'lodash/isObject';
+
+export type DeepPartial<T> = {
+  [P in keyof T]?: T[P] extends Record<number | string, unknown> ? DeepPartial<T[P]> : T[P];
+};
+
+const isPartialMatchObjectKeys = <ActualObject>({
+  objectToMatch,
+  objectToInspect,
+}: {
+  objectToMatch: DeepPartial<ActualObject>;
+  objectToInspect: ActualObject;
+}): boolean => {
+  const matchResponses: boolean[] = [];
+
+  const matchObjectKeys = ({
+    innerObjectToMatch,
+    innerObjectToInspect,
+  }: {
+    innerObjectToMatch: DeepPartial<ActualObject>;
+    innerObjectToInspect: DeepPartial<ActualObject>;
+  }): void => {
+    for (const [key, valueToMatch] of Object.entries(innerObjectToMatch)) {
+      const valueToInspect = innerObjectToInspect[key as keyof DeepPartial<ActualObject>];
+
+      if (innerObjectToInspect.hasOwnProperty(key)) {
+        if (
+          valueToMatch === null ||
+          valueToMatch === undefined ||
+          valueToMatch === '' ||
+          Array.isArray(valueToMatch) ||
+          // the condition checks if the "valueToMatch" is not of type object then "valueToMatch" type should be same as type of "valueToInspect"
+          (!(valueToMatch instanceof Object) && typeof valueToMatch !== typeof valueToInspect)
+        ) {
+          // this is an invalid case, so we log error
+          console.error(
+            `[isPartialMatchObjectKeys]: Unexpected value: ${JSON.stringify(
+              valueToMatch,
+            )} of type ${typeof valueToMatch} for key: ${key}`,
+          );
+          matchResponses.push(false);
```

Comment:
Why not just break the recursion early here itself and return `false`?

Reply:
it's a recursion inside an iteration. So it doesn't checks nested things. This is the first thing I had implemented and then learnt it the hard way 😄